symengine/polynomial.cpp

```diff
@@ -374,30 +374,31 @@ UnivariatePolynomial::UnivariatePolynomial(const RCP<const Symbol> &var,
     unsigned int deg = 0;
     for (unsigned int i = 0; i < v.size(); i++) {
         if (v[i] != 0) {
-            dict_[i] = v[i];
+            dict_.dict_[i] = v[i];
             deg = i;
         }
     }
     degree_ = deg;
```

Comment:
A check must be added here, to see if the `var` is `""`, then the vector must be of size 1 (i.e. a constant polynomial)